### DISCUSSION TITLE
Error decrypting message: No public key encrypted session key packet found.

### DISCUSSION THREAD

**Discussion by EvanJWeiss** [openpgpjs/openpgpjs#1633] [Q&A]
Nodejs version 18.x
OpenPGP version 5.7.0

I'm getting an error when decrypting an encrypted message. The message was encrypted using the example [here](https://www.npmjs.com/package/openpgp#encrypt-and-decrypt-string-data-with-pgp-keys), and here is the code I'm using to decrypt:

```js
const pgp_private_key = "<armored private key>"
const password = "<passphrase used when creating key pair>"
const encrypted_data = "<armored message from the encryption step>"

const privateKey = await openpgp.decryptKey({
    privateKey: await openpgp.readPrivateKey({armoredKey: pgp_private_key}),
    passphrase: password
});

const message = await openpgp.readMessage({
    armoredMessage: encrypted_data
});

const {data: decrypted, signatures} = await openpgp.decrypt({
    message: message,
    decryptionKeys: privateKey
});

return decrypted;
```

This is the error I receive with this code: 
`Error decrypting message: No public key encrypted session key packet found.`

I've also tried adding additional configurations to the decrypt step according to what little I could find regarding this issue.

```js
const {data: decrypted, signatures} = await openpgp.decrypt({
    message: message,
    decryptionKeys: privateKey,
    config: {
        allowInsecureDecryptionWithSigningKeys: true,
        allowInsecureVerificationWithReformattedKeys: true
    }
});
```

This error sometimes appears if the message is encrypted but not used for several hours or days (I have been unable to find a specific time frame). This would make sense for a session related issue where the session can expire. But nowhere in my encryption or decryption code am I setting a session value or checking it. The session keys parameter in the decrypt function list sessionKeys as optional as well, suggesting they are not required by default, so I am unsure on how to proceed. Additionally, due to a lack of documentation and others reporting this same error, I have not tried any other solutions. Other questions seem to reference this error: `Session key decryption failed`, and I haven't found any that state the session key packet was not found altogether.

Any help with this would be greatly appreciated!

**Comment by twiss** [comment]
Hey :wave: The only time-related factor should be whether the key is expired, but that wouldn't cause this error message, I think. Or possibly if you have stored the message on a rapidly-deteriorating hard drive? :sweat_smile:

Do you have an example message that fails to decrypt, possibly with an example test key?

And/or, could you try adding `ignoreUnsupportedPackets: false` to the config to see if you get a different / more helpful error message?

**Reply by EvanJWeiss** [reply]
Thanks for the reply! 

That's what I was thinking too, which is why I was so confused. Thank you for clarifying this though. Luckily, my hard drives are in working condition (I'm pretty sure 😂).

I will test this solution and possibly try creating a new key to see if something happened during creation. I will update this discussion with my findings as soon as I can.

**Comment by EvanJWeiss** [comment]
@twiss I generated a new RSA key-pair and it is working again. Knowing now that there is nothing time related to the encrypted message itself, I suspect I won't have any further issues 🤞. For the record, the `ignoreUnsupportedPackets` solution did not work for my situation, though I suspect with an invalid key that makes sense. Thanks for your help!

**Reply by twiss** [reply]
Alrighty! Was the broken key generated by OpenPGP.js or another implementation?
